perla-verde-traveler/src/pages/login.jsx

```diff
@@ -0,0 +1,68 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useUser } from "../provider/user_context";
+
+export default function LoginPage() {
+    const { login } = useUser();
+    const navigate = useNavigate();
+    const [userName, setUserName] = useState("");
+    const [password, setPassword] = useState("");
+    const [errorMessage, setErrorMessage] = useState("");
+
+    const handleLogin = (e) => {
+        e.preventDefault();
+        const mockUsers = [
+            { user_name: "pepito", password: "1234", first_name: "Pepito", last_name: "Perez" },
+            { user_name: "pascual", password: "1234", first_name: "Pascual", last_name: "Rodriguez" },
+            { user_name: "falcao", password: "1234", first_name: "Falcao", last_name: "Garcia" },
+        ];
+
+        const user = mockUsers.find(u => u.user_name === userName && u.password === password);
+
+        if (user) {
+            login(user);
+            setErrorMessage("");
+            navigate("/");
+        } else {
+            setErrorMessage("Usuario o contraseña incorrectos.");
+        }
+    };
```

Comment:
Podemos refactorizarlo así? para tener más entendimiento de que hace cada cosa y respetamos un poco el principio **SRP**

```suggestion
const mockUsers = [
    { user_name: "pepito", password: "1234", first_name: "Pepito", last_name: "Perez" },
    { user_name: "pascual", password: "1234", first_name: "Pascual", last_name: "Rodriguez" },
    { user_name: "falcao", password: "1234", first_name: "Falcao", last_name: "Garcia" },
];

const findUser = (userName, password) => {
    return mockUsers.find(u => u.user_name === userName && u.password === password);
};

const handleLogin = (e) => {
    e.preventDefault();

    const user = findUser(userName, password);

    if (user) {
        login(user);
        setErrorMessage("");
        navigate("/");
    } else {
        setErrorMessage("Usuario o contraseña incorrectos.");
    }
};
```

Reply:
Además, el mockUsers, lo puedes meter en un archivo de constants/mockUsers

Así lo podemos reutilizar cuando vayamos enlistar usuarios o algo por el estilo.